src/OAuthClient/HoldRequestClient.php

```diff
@@ -38,13 +42,13 @@ public static function validateRequestId($holdRequestId)
     public static function validateProcessed($processed)
     {
         if (!isset($processed)) {
-            throw new APIException(
-                'Processed flag not set',
-                'Processed flag is not set.',
-                0,
+            throw new NonRetryableException(
+                'Not Acceptable: Processed flag not set',
+                'Not Acceptable: Processed flag is not set.',
```

Comment:
Same as above for exception params; second param can be null instead of doubling the message.